### DISCUSSION TITLE
Why MoreMcmeta stutters my entire framerate?

### DISCUSSION THREAD

**Discussion by Neekodesu** [MoreMcmeta/core#83] [Help Wanted]
I can run minecraft with 80+ mods at 200+ fps with Bliss Shaders + Distant Horizons, but as soon as I load MoreMcmeta (since i do really want the emissive-plugin, fps/frametimes/everything becomes laggy unplayable.
I'll add the list of mods im using so you guys can test what's going on but, I have no idea.
Mods:
  [✔] AmbientSounds_FABRIC_v6.0.1_mc1.20.1
  [✔] animatica-0.6+1.20
  [✔] appleskin-fabric-mc1.20.1-2.5.1
  [✔] architectury-9.2.14-fabric
  [✔] backpacked-fabric-1.20.1-2.2.7
  [✔] BadOptimizations-2.1.1
  [✔] balm-fabric-1.20.1-7.3.4
  [✔] bendy-lib-fabric-4.0.0
  [✔] betterbeds-1.3.0-1.19.3
  [✔] bettercombat-fabric-1.8.5+1.20.1
  [✔] betterfpsdist-fabric-1.20.1-4.3
  [✔] bettermounthud-1.2.2
  [✔] blur-3.1.0
  [✔] c2me-fabric-mc1.20.1-0.2.0+alpha.11.5
  [✔] chat_heads-0.12.2-fabric-1.20
  [✔] chatanimation-1.0.5
  [✔] chatsigninghider-fabric-1.20.1-1.0.5
  [✔] Chunky-1.3.146
  [✔] citresewn-1.1.5+1.20.1
  [✔] cloth-config-11.1.118-fabric
  [✔] Clumps-fabric-1.20.1-12.0.0.4
  [✔] collective-1.20.1-7.61
  [✔] continuity-3.0.0-beta.5+1.20.1
  [✔] CreativeCore_FABRIC_v2.11.30_mc1.20.1
  [✔] cupboard-fabric-1.20.1-2.6
  [✔] Debugify-1.20.1+2.0
  [✔] DistantHorizons-fabric-2.0.4-a-dev-1.20.1
  [✔] emissive-plugin-1.20.1-2.0.4-fabric
  [✔] enhancedblockentities-0.9+1.20
  [✔] entity_model_features_fabric_1.20.1-2.0.2
  [✔] entity_texture_features_fabric_1.20.1-6.0.1
  [✔] entityculling-fabric-1.6.5-mc1.20.1
  [✔] fabric-api-0.92.2+1.20.1
  [✔] fabric-language-kotlin-1.11.0+kotlin.2.0.0
  [✔] fabricskyboxes-0.7.3+mc1.20.1
  [✔] fabrictailor-2.1.2
  [✔] fallingleaves-1.15.6+1.20.1
  [✔] FarmersDelight-1.20.1-2.1.1+refabricated
  [✔] fast-ip-ping-mc1.20.4-fabric-v1.0.1
  [✔] fastquit-3.0.0+1.20+
  [✔] ferritecore-6.0.1-fabric
  [✔] forcecloseloadingscreen-2.1.1
  [✔] FpsReducer2-fabric-1.20-2.5
  [✔] framework-fabric-1.20.1-0.6.16
  [✔] fsb-interop-1.3.6+mc1.20.1-build.50
  [✔] ImmediatelyFast-Fabric-1.2.17+1.20.4
  [✔] indium-1.0.30+mc1.20.4
  [✔] InventorySorter-1.9.0-1.20
  [✔] iris-1.7.0+mc1.20.1
  [✔] krypton-0.2.3
  [✔] lambdynamiclights-2.3.2+1.20.1
  [✔] lazydfu-0.1.3
  [✔] lithium-fabric-mc1.20.1-0.11.2
  [✔] make_bubbles_pop-0.2.0-fabric-mc1.19.4+
  [✔] memoryleakfix-fabric-1.17+-1.1.5
  [✔] midnightlib-fabric-1.4.1
  [✔] modernfix-fabric-5.18.0+mc1.20.1
  [✔] modmenu-7.2.2
  [✔] moreculling-1.20.4-0.24.0
  [✔] moremcmeta-1.20.1-4.4.7-fabric
  [✔] MouseTweaks-fabric-mc1.20-2.26
  [✔] noisium-fabric-2.0.3+mc1.20-1.20.1
  [✔] notenoughanimations-fabric-1.7.3-mc1.20.1
  [✔] notenoughcrashes-4.4.7+1.20.1-fabric
  [✔] nvidium-0.2.6-beta
  [✔] player-animation-lib-fabric-1.0.2-rc1+1.20
  [✔] reeses_sodium_options-1.7.2+mc1.20.1-build.101
  [✔] RoughlyEnoughItems-12.1.725-fabric
  [✔] rrls-4.0.6+mc1.20.1-fabric
  [✔] satin-1.14.0
  [✔] seriousplayeranimations-1.1.1
  [✔] shulkerboxtooltip-fabric-4.0.4+1.20.1
  [✔] sodium-extra-0.5.4+mc1.20.1-build.115
  [✔] sodium-fabric-0.5.8+mc1.20.1
  [✔] starlight-1.1.2+fabric.dbc156f
  [✔] threadtweak-fabric-1.20.2-0.1.1
  [✔] tooltipfix-1.1.1-1.20
  [✔] trashslot-fabric-1.20-15.1.0
  [✔] trinkets-3.7.2
  [✔] visuality-0.7.1+1.20
  [✔] Xaeros_Minimap_24.2.0_Fabric_1.20
  [✔] XaerosWorldMap_1.38.8_Fabric_1.20
  [✔] YetAnotherConfigLib-3.4.4+1.20.1-fabric
  [✔] youre-in-grave-danger-2.0.0-beta.13
  [✔] Zoomify-2.13.5+1.20.1

Removing moremcmeta-1.20.1-4.4.7-fabric actually reverse the stuttering.
Without Moremcmeta:
![image](https://github.com/MoreMcmeta/core/assets/47283957/db995679-f763-40c3-b743-2ceee836a66f)

With:
![image](https://github.com/MoreMcmeta/core/assets/47283957/2a60bb3c-bed3-466f-8a8a-2fe882adb877)

So yeah I don't know what could be causing this. 
![image](https://github.com/MoreMcmeta/core/assets/47283957/1f5d9eb6-2f48-4fa2-9209-5cbf9b66d3b5)

Currently 1.20.1 on Java 22 GraalVM.

EDIT: Same happening with only:
Mods:
  [✔] moremcmeta-1.20.1-4.4.7-fabric

So yeah I guess its not an incompatibility with a mod, is just MoreMcmeta being broken?
![image](https://github.com/MoreMcmeta/core/assets/47283957/dc5fbdce-8eef-454b-a4c4-d98fd0da5fc2)
With this framerate 100 fps feels like 15, I just can't play the game with it enabled, which sucks because continuity emissive textures is broken and yours actually works. Any fix or plan to do emissive-plugin non MoreMcmeta dependant?

**Comment by soir20** [comment]
Can you please try removing all resource packs and see if that improves the framerate? This is most likely an issue specific to a certain resource pack.

**Comment by Neekodesu** [comment]
Yea you're right, problem was emissive texture packs
For example https://modrinth.com/resourcepack/animated-ore

When using the emissive plugin, I should disable Continuity's emissive right? Since enabling/disabling does nothing

**Reply by soir20** [reply]
> When using the emissive plugin, I should disable Continuity's emissive right?

Yes, most likely they'd cause duplicate emissive overlays. Does the performance change at all when you enable or disable it?

> Yea you're right, problem was emissive texture packs

I think the issue is that this particular pack uses a lot of animated emissive overlays--this is a known issue but is a bit tricky to fix without causing incompatibility issues with Sodium, though it's still being worked on.
